stdlib/src/utils/string_slice.mojo

```diff
@@ -1047,24 +1047,21 @@ struct StringSlice[mut: Bool, //, origin: Origin[mut]](
                 offset += b_len
             return length != 0
 
-    fn splitlines[
-        O: ImmutableOrigin, //
-    ](self: StringSlice[O], keepends: Bool = False) -> List[StringSlice[O]]:
+    fn splitlines(
+        self: StringSlice, keepends: Bool = False
+    ) -> List[StringSlice[__type_of(self).origin]]:
         """Split the string at line boundaries. This corresponds to Python's
         [universal newlines:](
         https://docs.python.org/3/library/stdtypes.html#str.splitlines)
         `"\\r\\n"` and `"\\t\\n\\v\\f\\r\\x1c\\x1d\\x1e\\x85\\u2028\\u2029"`.
 
-        Parameters:
-            O: The immutable origin.
-
         Args:
             keepends: If True, line breaks are kept in the resulting strings.
 
         Returns:
             A List of Strings containing the input split by line boundaries.
         """
-
+        alias O = __type_of(self).origin
```

Comment:
**Suggestion** I wonder if we can remove this and use `StringSlice[self.origin]`, or just `List[Self]` below?